### PR TITLE
Ports gun attachments

### PR DESCRIPTION
**Pull request by Pariah919**
Heavily based off the old 2018 CM code with some changes. But yeah. Here you go. Its unfinished player facing wise and literally nothing uses it so far.

### CODE REVIEW

**Comment by Vizoplez** [comment]
Incredibly based? Compiled passed? 

Merging now.

**Comment by Kaksisilma** [comment]
Good PR premise, but there's a ton of check failures on attachments.dm

File: code/modules/projectiles/guns/attachments.dm [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+		accuracy += R.accuracy_modifier
+		screen_shake -= R.screen_shake_mod
+		bulk += R.bulk_mod
+		acc_mod_base +=R.acc_power_mod
+
+
+/obj/item/attachable/verticalgrip
+	name = "vertical grip"
+	desc = "A vertical foregrip that offers less loss of accuracy while moving."
+	icon_state = "grip"
```
Comment (Anton-Kr) [inline code/modules/projectiles/guns/attachments.dm]: All of the icon states are incorrect.

File: code/modules/projectiles/guns/attachments.dm [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	desc = "A angled foregrip that faster movement speed while firing."
+	icon_state = "grip"
+	attach_icon = "grip_a"
+	slot = "under"
+	bulk_mod = -2 // removes all rifle bulk
+
+/obj/item/attachable/holosight
+	name = "holosight"
+	desc = "Like a weird red dot sight. A zero magnification optic that offers faster, and more accurate target aquisition."
+	icon_state = "holosight"
```
Comment (Anton-Kr) [inline code/modules/projectiles/guns/attachments.dm:175]: This iconstate is missing.

File: code/modules/projectiles/guns/attachments.dm [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+	attach_icon = "holosight_a"
+	slot = "rail"
+	accuracy_modifier = 6
+
+// A stock that is unremoveable, this is used to make the gun more centered on the inventory and the ground!
+
+/obj/item/attachable/unremoveable_stock
+	name = "stock" // change this because it'll be the only seeable thing playerfacing, wise.
+	desc = "If you see this, an admin spawned in an irremoveable stock or someone was dumb - MJP."
+	icon_state = "holosight"
```
Comment (Anton-Kr) [inline code/modules/projectiles/guns/attachments.dm]: ditto

**Review by Anton-Kr — CHANGES_REQUESTED** [review]
Incorrect icon states.

File: code/modules/projectiles/guns/projectile/automatic.dm [hunk 4]
```diff
@@ -25,11 +25,18 @@
...
 
-// full auto versoin, as an example.
+// full auto versoin, as an example. NOW WITH ATTACHMENT EXAMPLE TOO!
 
 /obj/item/weapon/gun/projectile/automatic/fullauto
 	name = "prototype SMG"
 	desc = "A protoype lightweight, fast firing submachine gun. With a fun switch."
+	// Please for the love of whatever is up there, use this format for your list!
+	attachable_allowed = list(
+		/obj/item/attachable/verticalgrip,
```
Comment (Anton-Kr) [inline code/modules/projectiles/guns/projectile/automatic.dm:35]: Not sure if you intended for this to actually be used in game rather then the example, but if you are the offset is off and looks a biiit weird

**Review by Anton-Kr — CHANGES_REQUESTED** [review]
Other issue -> sprites of attachments disappear on the sprite on the firearm changing (ie ammo running out, mag removed, etc)

**Comment by Anton-Kr** [comment]
This ready for review @Pariah919 ?

File: code/modules/projectiles/guns/projectile/automatic.dm (hunk 4 above)
Comment (Pariah919) [reply code/modules/projectiles/guns/projectile/automatic.dm:35]: Sorry for being literally a week late but I've been busy the last week, but no the fullauto smg is more of an "example piece" for people to guideline themselves on how a future guns code should look like ~~or copy paste it as an example LMAO~~ for new guns.

**Comment by ProbablyCarl** [comment]
Marking as a draft until the author states this is ready for review.

**Review by Anton-Kr — APPROVED** [review]

**Pull request merged**